main/saturn-node.js

```diff
@@ -236,11 +236,14 @@ function forwardChunkFromSaturn (chunk, log) {
  * @param {string} text
  */
 function appendToChildLog (text) {
-  childLog += text
+  childLog.push(...text
     .trimEnd()
     .split(/\n/g)
-    .map(line => `[${new Date().toLocaleTimeString()}] ${line}\n`)
-    .join('')
+    .map(line => `[${new Date().toLocaleTimeString()}] ${line}`)
+  )
+  while (childLog.length > 100) {
+    childLog.shift()
+  }
```

Comment:
I makes me wonder what are the performance implication of this solution. 
- When we are adding a single line only, then calling `shift` once could be the most performant solution :+1:
- When we are adding multiple lines, we will repeatedly call `shift` as many times as many new lines were read from the L2 node. Could this become a performance issue? 🤔  

We can use `slice` instead, but that's creating a new copy of the array, which may have its own performance issues.

```
❯ node
Welcome to Node.js v16.16.0.
Type ".help" for more information.
> [1,2,3,4,5].slice(-2)
[ 4, 5 ]
```

```js
  childLog = childLog.slice(-100)
```

I think the best solution is to use [`splice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) and carefully calculate the offsets of items to delete.

```suggestion
  childLog.splice(0, childLog.length - 100)
```

WDYT?

I am debating with myself whether it's important or not to think about performance here. Your current solution is much easier to understand.

Reply:
Example from REPL calling `splice` to keep only last 3 items:

```
❯ node
Welcome to Node.js v16.16.0.
Type ".help" for more information.
> x = [1, 2, 3, 4, 5]
[ 1, 2, 3, 4, 5 ]
> x.splice(0, x.length - 3)
[ 1, 2 ]
> x
[ 3, 4, 5 ]
```

Reply:
I actually started with `slice`, but then switched to `shift` because of a micro benchmark:

```
> console.time('shift');{for(let i=0;i<1000;i++){const arr = new Array(105).fill();arr.shift();arr.shift();arr.shift();arr.shift();arr.shift()}}console.timeEnd('shift')
shift: 1.458ms
undefined
> console.time('slice');{for(let i=0;i<1000;i++){const arr = new Array(105).fill();arr.slice(-100)}}console.timeEnd('slice')
slice: 2.122ms
```

It's probably not a good benchmark (because of all the array allocations), but that was enough of a signal for me. Now that you brought up `splice` however, that is faster again:

```
> console.time('splice');{for(let i=0;i<1000;i++){const arr = new Array(105).fill();arr.splice(0, 5)}}console.timeEnd('splice')
splice: 0.943ms
```

I was wondering about performance here as well, it likely doesn't matter but since we both wondered let's just pick the faster implementation.